src/Functions/IFunction.cpp

```diff
@@ -203,6 +204,31 @@ ColumnPtr IExecutableFunction::defaultImplementationForNulls(
     return nullptr;
 }
 
+ColumnPtr IExecutableFunction::defaultImplementationForNothing(
+    const ColumnsWithTypeAndName & args, const DataTypePtr & result_type, size_t input_rows_count) const
+{
+    if (!useDefaultImplementationForNothing())
+        return nullptr;
+
+    bool is_nothing_type_presented = false;
+    for (const auto & arg : args)
+        is_nothing_type_presented |= isNothing(arg.type);
+
+    if (!is_nothing_type_presented)
+        return nullptr;
+
+    if (!isNothing(result_type))
+        throw Exception(
+            ErrorCodes::LOGICAL_ERROR,
+            "Function {} with argument with type Nothing and default implementation for Nothing "
+            "is expected to return result with type Nothing, got {}",
+            getName(),
+            result_type->getName());
+
+    return ColumnConst::create(ColumnNothing::create(1), input_rows_count);
```

Comment:
No, this should not be possible.
Maybe it's ok to have an empty `ColumnNothing`, but it can't contain any value.

Reply:
I think it would be safer to:
* throw if input_rows_count > 0
* return just empty ColumnNothing otherwise

Reply:
Fixed